src/include/sof/schedule/dp_schedule.h

```diff
@@ -0,0 +1,100 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Marcin Szkudlinski
+ */
+
+#ifndef __SOF_SCHEDULE_DP_SCHEDULE_H__
+#define __SOF_SCHEDULE_DP_SCHEDULE_H__
+
+#include <sof/schedule/task.h>
+#include <sof/trace/trace.h>
+#include <user/trace.h>
+#include <stdint.h>
+
+/**
+ *
+ * DP scheduler is a scheduler that creates a separate preemptible Zephyr thread for each SOF task
+ *   There's only one instance of DP in the system, however, threads can be assigned and pinned
+ *   to any core in the system for its execution, there's no SMP processing.
+ *
+ * The task execution may be delayed and task may be re-scheduled periodically
+ * NOTE: delayed start and rescheduling takes place in sync with LL scheduler, meaning the
+ *   DP scheduler is triggered as the last task of LL running on a primary core.
+ *   That implies a limitation: LL scheduler MUST be running on primary core in order to have
+ *   this feature working.
+ *   It is fine, because rescheduling is a feature used for data processing when a pipeline is
+ *   running.
+ *
+ * Other possible usage of DP scheduler is to schedule task with DP_SCHEDULER_RUN_TASK_IMMEDIATELY
+ *   as start parameter. It will force the task to work without any delays and async to LL.
+ *   This kind of scheduling may be used for tasks like IPC.
+ *
+ * Task run() may return:
+ *  SOF_TASK_STATE_RESCHEDULE - the task will be rescheduled as specified in scheduler period
+ *				note that task won't ever be rescheduled if LL is not running
+ *  SOF_TASK_STATE_COMPLETED  - the task will be removed from scheduling, but not terminated
+ *				calling schedule_task will add the task to processing again
+ *				task_complete() will be called
+ *  SOF_TASK_STATE_CANCEL     - the task will be cancelled permanently,
```

Comment:
self review: DP should be able to reschedule the task after canceling. 
To remove a task from periodic scheduling task_cancel must be called (of SOF_TASK_STATE_CANCEL/ SOF_TASK_STATE_COMPLETED returned trom task_run). Anyway, there's no reason why the task is terminated when cancelled and can't be scheduled again. To be fixed.